repository.go

```diff
@@ -8,122 +8,159 @@ import (
 	_ "github.com/lib/pq"
 )
 
-const (
-	host     = "localhost"
-	port     = 5432
-	user     = "postgres"
-	password = "1234"
-	dbname   = "test"
-)
-
-func getCart() {
-
-	psqlInfo := fmt.Sprintf("host=%s port=%d user=%s "+
-		"password=%s dbname=%s sslmode=disable",
-		host, port, user, password, dbname)
-	db, err := sql.Open("postgres", psqlInfo)
-	if err != nil {
-		panic(err)
-	}
-	defer db.Close()
+func getItem(db *sql.DB) {
+	sqlStatement :=
+		"select item_id, item_name, item_value " +
+			"from public.item " +
+			"where item_id = ?; "
+	sqlStatement = strings.Replace(sqlStatement, "?", "1", 1)
+	fmt.Println(sqlStatement)
 
-	err = db.Ping()
-	if err != nil {
+	var item Item
+	row := db.QueryRow(sqlStatement)
+	switch err := row.Scan(&item.id, &item.name, &item.value); err {
+	case sql.ErrNoRows:
+		fmt.Println("No rows were returned!")
+	case nil:
+		fmt.Println(item)
+	default:
 		panic(err)
```

Comment:
I know this is not a new line in this PR, but you should never use `panic()` in a function which is not `main` or initialization code. If you app is already running and because a call to this endpoint you reach panic. it will kill your entire system (other healthy endpoints). Instead, you should just return the error and control it on the caller of this function